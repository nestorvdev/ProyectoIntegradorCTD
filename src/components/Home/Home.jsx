import Categories from "../Categories/Categories.jsx";
import SearchBlock from "../SearchBlock/SearchBlock";
import Cards from "../Cards/Cards";
import Spinner from "../spinner/Spinner";
import { Modal } from 'react-responsive-modal';
import 'react-responsive-modal/styles.css';
import ModalProductoErroneo from "../Administrator/Product/ModalProductSucceed";
import { useState, useEffect} from "react";


export default function Home(props) {

  const [borrarCity, setBorrarCity] = useState(false);
  const [borrarDate, setBorrarDate] = useState(false);
 
  useEffect(() => {
    props.setActiveCreate(false)
    props.setActiveLogin(false)
  }, [props])


  if (sessionStorage.getItem("log") === "false" && !props.loading) {
    sessionStorage.removeItem("iniciales")
    sessionStorage.removeItem("name")
    sessionStorage.removeItem("surname")
  }

  const closeModal = () => {
    props.setProductoErroneo(false)
    sessionStorage.removeItem("productoErroneo")
  };

  return (
    <>
      {props.loading ? (
        <Spinner />
      ) : (
        <>
          <SearchBlock borrarCity={borrarCity} setBorrarCity={setBorrarCity} borrarDate={borrarDate} setBorrarDate={setBorrarDate} handleSearch={props.handleSearch} handleCity={props.handleCity} city={props.city} setStartDate={props.setStartDate} setEndDate={props.setEndDate} startDate={props.startDate} endDate={props.endDate} />
          <Categories category={props.category} handleCategory={props.handleCategory} />
          <Cards setBorrarCity={setBorrarCity} setBorrarDate={setBorrarDate} setLastLocation={props.setLastLocation} category={props.category} handleClean={props.handleClean} search={props.search} city={props.city} clickBusqueda={props.clickBusqueda} favourite={props.favourite} clickSeeFavourites={props.clickSeeFavourites} setCategory={props.setCategory} setCity={props.setCity} setStartDate={props.setStartDate} setEndDate={props.setEndDate} propStartDate={props.startDate} propEndDate={props.endDate} setFavourite={props.setFavourite} setSearch={props.setSearch} />
          <Modal open={props.productoErroneo} onClose={closeModal} center>
            <ModalProductoErroneo title={"Error"} message={"El producto no existe."} closeModal={closeModal} icon="X"/>
          </Modal>
        </>
      )}
    </>
  );
}
