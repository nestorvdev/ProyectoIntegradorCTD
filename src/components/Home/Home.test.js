import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Home from "./Home";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <Home/>", () => {
    let wrapper;
    let setActiveCreate = jest.fn(); //Simula la funcion del set
    let setActiveLogin = jest.fn() //Simula la funcion del set

    beforeEach(() => {
        wrapper = shallow(<Home setActiveCreate={setActiveCreate} setActiveLogin= {setActiveLogin}/>)
    });

    it("Deberia mostrar <Home/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    
})