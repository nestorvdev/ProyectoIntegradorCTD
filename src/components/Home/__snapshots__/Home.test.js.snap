// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <Home/> Deberia mostrar <Home/> correctamente 1`] = `
<Fragment>
  <SearchBlock
    borrarCity={false}
    borrarDate={false}
    setBorrarCity={[Function]}
    setBorrarDate={[Function]}
  />
  <Categories />
  <Cards
    setBorrarCity={[Function]}
    setBorrarDate={[Function]}
  />
  <ForwardRef
    center={true}
    onClose={[Function]}
  >
    <ProductModalSucceed
      closeModal={[Function]}
      icon="X"
      message="El producto no existe."
      title="Error"
    />
  </ForwardRef>
</Fragment>
`;
